### PR TITLE
Add permissions to EditChannel builder

### PR DESCRIPTION
**Pull request by pcbzzz**

### CODE REVIEW

**Review by arqunis — APPROVED** [review]

File: src/builder/edit_channel.rs [hunk 1]
```diff
@@ -115,4 +119,59 @@ impl EditChannel {
...
+    /// use serenity::model::permissions::Permissions;
+    ///
+    /// // Assuming a guild have already been bound.
+    /// let permissions = Some(PermissionOverwrite {
+    ///     allow: Permissions::READ_MESSAGES,
+    ///     deny: Permissions::SEND_TTS_MESSAGES,
+    ///     kind: PermissionOverwriteType::Member(UserId(1234)),
+    /// });
+    ///
+    /// guild.create_channel(http, |c| {
```
Comment (arqunis) [inline src/builder/edit_channel.rs]: This should show an example of editing a channel, not creating a new one.

File: src/builder/edit_channel.rs [hunk 2]
```diff
@@ -115,4 +119,59 @@ impl EditChannel {
...
+    /// # use std::sync::Arc;
+    /// #
+    /// # async fn run() -> Result<(), Box<dyn std::error::Error>> {
+    /// #     let http = Arc::new(Http::default());
+    /// #     let mut guild = GuildId(0).to_partial_guild(&http).await?;
+    /// use serenity::model::channel::{PermissionOverwrite, PermissionOverwriteType};
+    /// use serenity::model::id::UserId;
+    /// use serenity::model::permissions::Permissions;
+    ///
+    /// // Assuming a guild have already been bound.
```
Comment (arqunis) [inline src/builder/edit_channel.rs]: "guild" is the subject of the sentence, which is in third person.
```suggestion
    /// // Assuming a guild has already been bound.
```

This typo is also present in the example for [`CreateChannel::permissions`][perms], could you fix it there too?

[perms]: https://github.com/serenity-rs/serenity/blob/current/src/builder/create_channel.rs#L108

**Review by arqunis — CHANGES_REQUESTED** [review]

File: src/builder/edit_channel.rs (hunk 2 above)
Comment (pcbzzz) [reply src/builder/edit_channel.rs]: Fixed both. Considering it was a direct copy of that function I'm not surprised, I noticed the `create_channel` after I submitted and just hoped it was overlooked 😜

File: src/builder/edit_channel.rs [hunk 3]
```diff
@@ -115,4 +119,59 @@ impl EditChannel {
...
+    /// use serenity::model::permissions::Permissions;
+    ///
+    /// // Assuming a guild has already been bound.
+    /// let permissions = Some(PermissionOverwrite {
+    ///     allow: Permissions::READ_MESSAGES,
+    ///     deny: Permissions::SEND_TTS_MESSAGES,
+    ///     kind: PermissionOverwriteType::Member(UserId(1234)),
+    /// });
+    ///
+    /// guild.edit_channel(http, |c| {
```
Comment (arqunis) [inline src/builder/edit_channel.rs]: Guild models don't have an `edit_channel` method (though they should). Use [`GuildChannel::edit`](https://serenity-rs.github.io/serenity/current/serenity/model/channel/struct.GuildChannel.html#method.edit).

**Review by arqunis — CHANGES_REQUESTED** [review]

**Review by arqunis — APPROVED** [review]

**Pull request merged**